src/components/Remark.tsx

```diff
@@ -0,0 +1,101 @@
+import { u8aToHex } from '@polkadot/util';
+import React, { useState } from 'react';
+import { Button, Container, Input } from 'semantic-ui-react';
+import styled from 'styled-components';
+
+import { useApiSourcePromiseContext } from '../contexts/ApiPromiseSourceContext';
+import { useApiTargetPromiseContext } from '../contexts/ApiPromiseTargetContext';
+import useLoadingApi from '../hooks/useLoadingApi';
+interface Props {
+  className?: string;
+  targetChain: string;
+}
+
+const Remark = ({ className, targetChain }: Props) => {
+  const { api: sourceApi } = useApiSourcePromiseContext();
+  const { api: targetApi } = useApiTargetPromiseContext();
+
+  const areApiLoading = useLoadingApi();
+  const [remarkInput, setRemarkInput] = useState('0x');
+  const [executionStatus, setExecutionStatus] = useState('');
+
+  const onChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setExecutionStatus('');
+    setRemarkInput(event.target.value);
+  };
+
+  async function sendMessageRemark() {
+    const keyring = new Keyring({ type: 'sr25519' });
+    const account = keyring.addFromUri('//Alice');
+
+    const remarkCall = await targetApi.tx.system.remark(remarkInput);
+    const remarkInfo = await sourceApi.tx.system.remark(remarkInput).paymentInfo(account);
+
+    const weight = remarkInfo.weight.toNumber();
+
+    const call = remarkCall.toU8a();
+
+    const lane_id = new Uint8Array(8);
+    const payload = {
+      call,
+      origin: {
+        SourceAccount: account.addressRaw
+      },
+      spec_version: 1,
+      weight
+    };
+
+    // Ignoring custom types missed for TS for now.
+    // Need to apply: https://polkadot.js.org/docs/api/start/typescript.user
+    // @ts-ignore
+    const payloadType = sourceApi.registry.createType('OutboundPayload', payload);
+    // @ts-ignore
+    const messageFeeType = sourceApi.registry.createType('MessageFeeData', {
+      lane_id: '0x00000000',
+      payload: u8aToHex(payloadType.toU8a())
+    });
+
+    const estimatedFeeCall = await sourceApi.rpc.state.call<Codec>(
+      `To${targetChain}OutboundLaneApi_estimate_message_delivery_and_dispatch_fee`,
+      u8aToHex(messageFeeType.toU8a())
+    );
+
+    // @ts-ignore
+    const estimatedFeeType = sourceApi.registry.createType('Option<Balance>', estimatedFeeCall);
+    const estimatedFee = estimatedFeeType.toString();
+
+    const bridgeMessage = sourceApi.tx[`bridge${targetChain}MessageLane`].sendMessage(lane_id, payload, estimatedFee);
+
+    const nonceCall = await sourceApi.rpc.system.accountNextIndex(account.address);
+    const nonce = nonceCall.toNumber();
+    bridgeMessage.signAndSend(account, { nonce }).then(() => setExecutionStatus('Message delivered'));
+  }
+
+  if (!areApiLoading) {
+    return null;
+  }
+
+  return (
+    <Container className={className}>
+      <Input onChange={onChange} value={remarkInput} />
+      <Button onClick={sendMessageRemark}>Message Remark</Button>
+      {executionStatus !== '' && (
```

Comment:
this section is only for giving some feedback that the message was sent. It will be removed once working on the status steps detail.